src/analysis/special_functions/gamma.lean

```diff
@@ -1214,3 +1218,259 @@ end
+lemma Gamma_neg_nat_eq_zero (n : ℕ) : Gamma (-n) = 0 :=
+begin
+  induction n with n IH,
+  { rw [Gamma, nat.cast_zero, neg_zero, zero_re, sub_zero, nat.floor_one, Gamma_aux],
+    simp only [div_zero] },
+  { have A : -(n.succ : ℂ) ≠ 0,
+    { rw [neg_ne_zero, nat.cast_ne_zero],
+      apply nat.succ_ne_zero },
+    have : -(n:ℂ) = -↑n.succ + 1, by simp,
+    rw [this, complex.Gamma_add_one _ A] at IH,
+    contrapose! IH,
+    exact mul_ne_zero A IH }
+end
+
+theorem Gamma_mul_Gamma_one_sub (z : ℂ) : Gamma z * Gamma (1 - z) = π / sin (π * z) :=
+begin
+  have pi_ne : (π : ℂ) ≠ 0, from complex.of_real_ne_zero.mpr pi_ne_zero,
+  by_cases hs : sin (↑π * z) = 0,
+  { -- first deal with silly case z = integer
+    rw [hs, div_zero],
+    rw [←neg_eq_zero, ←complex.sin_neg, ←mul_neg, complex.sin_eq_zero_iff, mul_comm] at hs,
+    obtain ⟨k, hk⟩ := hs,
+    rw [mul_eq_mul_right_iff, eq_false_intro (of_real_ne_zero.mpr pi_pos.ne'), or_false,
+      neg_eq_iff_neg_eq] at hk,
+    rw ←hk,
+    cases k,
+    { rw [int.cast_of_nat, complex.Gamma_neg_nat_eq_zero, zero_mul] },
+    { rw [int.cast_neg_succ_of_nat, neg_neg, nat.cast_add, nat.cast_one, add_comm, sub_add_cancel',
+        complex.Gamma_neg_nat_eq_zero, mul_zero] } },
+  refine tendsto_nhds_unique ((Gamma_seq_tendsto_Gamma z).mul (Gamma_seq_tendsto_Gamma $ 1 - z)) _,
+  have : ↑π / sin (↑π * z) = 1 * (π / sin (π * z)), by rw one_mul, rw this,
+  refine tendsto.congr' ((eventually_ne_at_top 0).mp
+    (eventually_of_forall (λ n hn, (Gamma_seq_mul z hn).symm))) (tendsto.mul _ _),
+  { convert tendsto_coe_nat_div_add_at_top (1 - z), ext1 n, rw add_sub_assoc },
+  { have : ↑π / sin (↑π * z) = 1 / (sin (π * z) / π), by field_simp, rw this,
+    refine tendsto_const_nhds.div _ (div_ne_zero hs pi_ne),
+    rw [←tendsto_mul_iff_of_ne_zero tendsto_const_nhds pi_ne, div_mul_cancel _ pi_ne],
+    convert tendsto_euler_sin_prod z,
+    ext1 n, rw [mul_comm, ←mul_assoc] },
+end
+
+theorem Gamma_ne_zero {s : ℂ} (hs : ∀ m : ℕ, s ≠ -m) : Gamma s ≠ 0 :=
+begin
+  by_cases h_im : s.im = 0,
+  { have : s = ↑s.re,
+    { conv_lhs { rw ←complex.re_add_im s }, rw [h_im, of_real_zero, zero_mul, add_zero] },
+    rw [this, Gamma_of_real, of_real_ne_zero],
+    refine real.Gamma_ne_zero (λ n, _),
+    specialize hs n,
+    contrapose! hs,
+    rwa [this, ←of_real_nat_cast, ←of_real_neg, of_real_inj] },
+  { have : sin (↑π * s) ≠ 0,
+    { rw complex.sin_ne_zero_iff,
+      intro k,
+      apply_fun im,
+      rw [of_real_mul_im, ←of_real_int_cast, ←of_real_mul, of_real_im],
+      exact mul_ne_zero real.pi_pos.ne' h_im },
+    have A := div_ne_zero (of_real_ne_zero.mpr real.pi_pos.ne') this,
+    rw [←complex.Gamma_mul_Gamma_one_sub s, mul_ne_zero_iff] at A,
+    exact A.1 }
+end
+
+lemma Gamma_eq_zero_iff (s : ℂ) : Gamma s = 0 ↔ ∃ m : ℕ, s = -m :=
+begin
+  split,
+  { contrapose!, exact Gamma_ne_zero },
+  { rintro ⟨m, rfl⟩, exact Gamma_neg_nat_eq_zero m },
+end
+
+end complex
+
```

Comment:
Could you also deduce the real versions of the reflection formula, and of `Gamma_neg_nat_eq_zero` and friends?